assembly/assembly-ide-war/src/main/webapp/_app/IDE.html

```diff
@@ -54,12 +54,278 @@
+            this.loadCompilationMappingsPropertiesFile = function() {
+                var msg = "Unable to load compilation-mappings.properties";
+
+                try {
+                    var request = new XMLHttpRequest();
+
+                    request.onerror = request.abort = function() {
+                        console.error(msg);
+                    };
+
+                    request.onload = function() {
+                        if (request.status == 200) {
+                            Loader.determineIDEJavaScriptContentLength(request.responseText);
+                        } else {
+                            console.error(msg);
+                        }
+                    };
+
+                    var url = "/_app/compilation-mappings.properties?" + new Date().getTime();
+                    request.open("GET", url, true);
+                    request.send();
+                } catch (e) {
+                    console.error(msg, e);
+                }
+            };
+
+            /*
+             * Determine IDE java script file size
+             */
+            this.determineIDEJavaScriptContentLength = function(compilationMappingsProperties) {
+                var mappings = compilationMappingsProperties.split("\n");
+
+                try {
+                    for (var i = 0; i < mappings.length; i++) {
+                        var parts = mappings[i].split(" : ");
+
+                        if (parts[0] === Loader.ideJavaScriptFileName) {
+                            Loader.ideJavaScriptFileSize = parseInt(parts[1]);
+                            Loader.loadIDEJavaScript();
+                        }
+                    }
+                } catch (e) {
+                    console.error("Cannot determine size of IDE JavaScript", e);
+                }
+            };
+
+            /*
+             * Load content of IDE JavaScript
+             */
+            this.loadIDEJavaScript = function() {
+                var msg = "Unable to load IDE JavaScript";
+
+                try {
+                    var request = new XMLHttpRequest();
+
+                    request.onprogress = function(event) {
+                        var percentComplete = event.loaded / Loader.ideJavaScriptFileSize * 100;
+                        var rounded = Math.round(percentComplete);
+                        if (rounded > 100) {
+                            rounded = 100;
+                        }
+
+                        document.getElementById("ide-loader-progress-bar").style.width = rounded + "%";
+                    };
+
+                    request.onload = function() {
+                        if (request.status == 200) {
+                            Loader.injectIDEJavaScript();
+                        } else {
+                            console.error(msg);
+                        }
+                    };
+
+                    request.onerror = request.abort = function () {
+                        console.error(msg);
```

Comment:
do we have root cause ?

Reply:
I changed it a little bit. IDE javascript included on any error. The loading process must not be interrupted is errors occur.

Reply:
I mean, it would be helpful if root cause is displayed on javascript console

Reply:
Yes, understand. Fixing.

Reply:
Browser usually writes the error to the console. Do we need to duplicate it?

Reply:
if in console you've 100 errors, it might help if you know which error is associated to which call but up to you

Reply:
It must be before the line containing message "Unable to load IDE JavaScript".